var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -25,27 +25,42 @@ class Zlib(Package):
             description='Produce position-independent code (for shared libs)')
     variant('shared', default=True,
             description='Enables the build of shared libraries.')
+    variant('debug',   default=False,
+            description='Build debug code (-g -O0)')
     variant('optimize', default=True,
             description='Enable -O2 for a more optimized lib')
 
+    conflicts('+debug+optimize')
```

Comment:
Is this supported? I don't see how it would work looking through `_old_concretize`. Generally conflicts are specified like

```
conflicts('+debug', when='+optimize')
```

Reply:
Both are supported.